hg2git.py

```diff
@@ -132,7 +132,7 @@ def get_git_sha1(name,type='heads'):
   try:
     # use git-rev-parse to support packed refs
     ref="refs/%s/%s" % (type,name.decode('utf8'))
-    l=subprocess.check_output(["git", "rev-parse", "--verify", "--quiet", ref])
+    l=subprocess.check_output(["git", "rev-parse", "--verify", "--quiet", ref.encode('utf8')])
```

Comment:
@Zubastic, could you explain why the `ref.encode('utf8')` is needed? Are you running Python 2 or 3?

Reply:
@frej because when ref having non ascii chars in branch name it fail

Reply:
Isn't it redundant? `get_git_sha1()` is only called from `verify_heads()` and there `force_utf8()` will already have ensured that the branch name is utf-8.

Reply:
No, ref is not in utf-8 encoding and it fails when try to use git command.

Reply:
That doesn't make sense, either the branch name is utf-8 and everything is fine, or else you run with -be and `b=b.decode(branch_encoding).encode('utf8')` (which should have been replaced with a `force_utf8()`) on line 485 will make sure it its. Do you have a branch mapping file with a non-utf8 mapping?

Reply:
Sorry @frej I managed to miss the ping there.

I don't quite understand why, but on some systems (or Python versions?) the default encoding for `.encode()`, `.decode()` or opening a file is `ascii` and not `utf8`. For me here on Python 3.8 on Arch Linux with my locale configured how it is the default encoding is definitely `utf8`, but I've had bug reports against other projects of mine that demonstrate that this can't be assumed.

So the added `.encode('utf8')` here is actually telling Python to encode with utf8 to construct the command line, instead of the implicit `.encode()` (defaulting to ascii on some systems) that it will otherwise do.

I think this change is correct to prevent raising a `UnicodeEncodeError` for non-ascii branch names, and yep I think it should work for all Pythons.

Reply:
@chrisjbillington no worries, the wait was worth it as we now know what's going on.

As this fixes the problem in #233 and  @Zubastic's PR seems to have stalled, I'll make a patch doing the change, crediting @chrisjbillington  for the explanation and @Zubastic for discovering the fix.